sycl/test-e2e/KernelCompiler/kernel_compiler_sycl_jit.cpp

```diff
@@ -612,12 +562,33 @@ int test_warning() {
   return 0;
 }
 
+int test_no_visible_ids() {
+  namespace syclex = sycl::ext::oneapi::experimental;
+  using source_kb = sycl::kernel_bundle<sycl::bundle_state::ext_oneapi_source>;
+  using exe_kb = sycl::kernel_bundle<sycl::bundle_state::executable>;
+  sycl::queue q;
+  sycl::context ctx = q.get_context();
+
+  bool ok =
+      q.get_device().ext_oneapi_can_compile(syclex::source_language::sycl_jit);
+  if (!ok) {
```

Comment:
why don't we fail test as done in the other tests?
I think if we expect these tests to work on specific devices (that is usually managed by REQUIRES) we should report error if conditions are not met unexpectedly.

Reply:
Yes thanks for pointing this out, I copy-pasted from `test_warning` without noticing that it was returning 0. I checked offline with @jopperm and we decided to refactor the test a bit so that the check is done only once in main, also given how the test is structured there was no point in having `test_warning` optional so we removed that, I also removed the capability check in the tests and put in `main` so that it's done once.